jarvis_files/jarvis_cmd/src/jarvis/build.py

```diff
@@ -169,3 +172,89 @@ def build_all(ctx, d, lint, opts, not_build):
     print("Successfully built: {} of {} project(s).".format(
         num_projects - len(failed_projects), num_projects))
     return len(failed_projects)
+
+# Function that parses launch command
+def launch_dir(ctx, d, opts, ssh):
+    if d == "percep":
+        launch_percep(ctx, opts, ssh)
+    if d == "nav":
+        launch_nav(ctx, opts, ssh)
+    if d == "loc":
+        launch_loc(ctx, opts, ssh)
+    if d == "auton":
+        launch_auton(ctx, opts, ssh)  
+    return
+
+def get_process_id(name):
+
+    child = subprocess.Popen(['pgrep', '-f', name], stdout=subprocess.PIPE, shell=False)
+    response = child.communicate()[0]
+    return [int(pid) for pid in response.split()]
+
+def wait_for_click():
+    pid = get_process_id("xdotool selectwindow")
+    while pid:
+        pid = get_process_id("xdotool selectwindow")
+        time.sleep(.1)
+    return 
+
+def genTerminalLaunchCommand(script_address, ssh):
+    command = "gnome-terminal -- bash -c './"
+    command = command + script_address
+
+    if ssh:
+        return command + " ssh; $SHELL'"
+    else:
+        return command + "; $SHELL'"
+
+# Functions that build and execute auton subteam code
+def launch_percep(ctx, opts, ssh):
+    percep = 'jetson/percep'
+    l = 'True'
+    workspace_relative_address = "jarvis_files/jarvis_cmd/launchScripts/percep"
+    
+    build_dir(ctx, percep, l, opts)
+
+    ctx.run( genTerminalLaunchCommand(workspace_relative_address, ssh) )
+    
+    wait_for_click()
+    return
+
+def launch_nav(ctx, opts, ssh):
+    nav = 'jetson/nav'
+    l = 'True'
+    workspace_relative_address = "jarvis_files/jarvis_cmd/launchScripts/nav"
+    
+    build_dir(ctx, nav, l, opts)
+    
+    ctx.run( genTerminalLaunchCommand(workspace_relative_address, ssh) )
+
+    wait_for_click()
+    return
+
+def launch_loc(ctx, opts, ssh):
+    gps = 'jetson/gps'
+    filter = 'jetson/filter'
+    l = 'True'
```

Comment:
This should be called "use_linter" right?

Reply:
Sry. Must've been a typo